src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -315,6 +315,30 @@ public Money GetSentTransactionFee(uint256 transactionId)
 
             return inputsAmount - outputsAmount;
         }
+
+        /// <summary>
+        /// Finds the HD addresses for the public address.
+        /// </summary>
+        /// <remarks>
+        /// Returns an HDAddress.
+        /// </remarks>
+        /// <param name="externalAddress">An address.</param>
+        /// <param name="accountFilter">An optional filter for filtering the accounts being returned.</param>
+        /// <returns>HD Address</returns>
+        public HdAddress FindHDAddressByExternalAddress(string externalAddress, Func<HdAccount, bool> accountFilter = null)
```

Comment:
- You can rename this to `GetAddress(...)`.
- Please return null if no address is found.

Have a look at the `public HdAccount GetAccount(string accountName)` method.